contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -60,9 +60,25 @@ def execute(self):
           lib_binary_map[vt.target] = os.path.join(gopath, 'pkg', self.goos_goarch,
                                                    vt.target.import_path + '.a')
 
+  @classmethod
+  def _split_build_flags(cls, build_flags):
+    """Visible for testing"""
+    return safe_shlex_split(build_flags)
+
   def _go_install(self, target, gopath):
-    build_flags = re.sub(r'^"|"$', '', self.get_options().build_flags)
-    args = safe_shlex_split(build_flags) + [target.import_path]
+    """Create and execute a `go install` command.
+
+    Build flags can be defined as globals (in `pants.ini`), as arguments to a Target, and
```

Comment:
I think it's worth extracting `_calculate_build_flags` or somesuch as its own method. It's meaty logic and arguably obscures `self.go_dist...` which is the proper highlight of the method.